### PR TITLE
fix IMAGE=2 failed with NOOPT=1

### PR DESCRIPTION
**Pull request by chenyuxyz**
even with NOOPT, we need `required_optimizations` to set up shape for image dtypes

### CODE REVIEW

File: tinygrad/ops.py [hunk 1]
```diff
@@ -325,6 +325,9 @@ def get_program():
           timed = sorted([(nm, tk, time_linearizer(tk, test_rawbuffers, allow_test_size=False, disable_cache=True, clear_l2=True)) for nm, tk in lins], key=lambda x: x[2])
           if DEBUG >= 1: print("  <  ".join(f"{nm:6s} : {lin.colored_shape(30, dense=True)} : {tm*1e6:8.2f} us" for nm, lin, tm in timed))
           k = timed[0][1]
+      else:
+        pass
```
Comment (geohot) [inline tinygrad/ops.py:329]: I don't think you want pass here

Comment (chenyuxyz) [reply tinygrad/ops.py]: that was to demo that CI would catch this

**Comment by chenyuxyz** [comment]
not entirely sure what's happening but the triton test segfault might be real, it also happened on my local fork. let me investigate

**Comment by chenyuxyz** [comment]
yeah i think #2214  fixed it. ran it 10 times in my local fork and it all passed https://github.com/chenyuxyz/tinygrad/actions/workflows/test.yml

**Pull request merged**